src/hash_table.h

```diff
@@ -75,6 +75,7 @@ struct ht_rec {
 struct hash_table {
     uint32_t used;        /* number of values stored in the hash table (filled records) */
     uint32_t size;        /* always holds 2^x == size (is power of 2), actually number of records allocated */
+    uint32_t deleted;     /* number of delete records in the hash table */
```

Comment:
Please stick to the terminology, this should be `invalid`. The comment can mostly stay, the meaning is fine, just some explanation to add that deleted records are invalid.

Reply:
lastest code is :
uint32_t invalid;     /* number of invaild records in the hash table (deleted records) */